File: 283-move-zeroes/move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        i = 0
        zero_counter = 0
        while i < len(nums) - zero_counter:
            if nums[i] == 0:
                nums.pop(i)
                nums.append(0)
                zero_counter += 1
            else:
                i += 1
        

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int slow = 0;

        for (int fast = 0; fast < nums.length; fast++) {
            if (nums[fast] != 0 && nums[slow] == 0) {
                // switch slow and fast ptr
                int temp = nums[slow];
                nums[slow] = nums[fast];
                nums[fast] = temp;
            }

            if (nums[slow] != 0) {
                slow++;
            }
            // System.out.println(Arrays.toString(nums));
        }

    }
}